core/src/main/java/brooklyn/util/internal/ssh/ShellAbstractTool.java

```diff
@@ -105,7 +105,7 @@ protected static Boolean hasVal(Map<String,?> map, ConfigKey<?> keyC) {
     public static <T> T getOptionalVal(Map<String,?> map, ConfigKey<T> keyC) {
         if (keyC==null) return null;
         String key = keyC.getName();
-        if (map!=null && map.containsKey(key)) {
+        if (map!=null && map.containsKey(key) && map.get(key) != null) {
```

Comment:
I'll go with this and merge it.

However, I'm slightly hesitant: in some places, it's nice to be able to explicitly supply null so that it clears the value rather than it using the default. For example, in the use of JcloudsLocation I want to be able to set the privateKeyFile to none, rather than it picking the default of "~/.ssh/id_rsa". But I could probably still do that with a blank string. So not sure if there are any valid use-cases for explicitly setting null.